### DISCUSSION TITLE
Type error: Exported variable 'X' has or is using name 'PrdStore' from external module

### DISCUSSION THREAD

**Discussion by gitsunmin** [pmndrs/jotai#2717] [Bug report]
### Bug Description

There was no problem with the code below when `jotai 2.8.3` was used.

```ts
import { atom, createStore } from 'jotai';

export const xAtom = atom<Array<{ id: string; }>>([]);
export const xStore = createStore();

```

However, after upgrading to `jotai 2.9.3`, I received the following error message when I did "Next build".

```shell
Type error: Exported variable 'xStore' has or is using name 'PrdStore' from external module
```

When I checked, it seems to be caused because `Store` type is not "export".

```ts
/* export */ type Store = PrdStore | (PrdStore & DevStoreRev4);
```

What do you think about adding "export"?

### Reproduction Link

https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgQxhEAaOBjKBTVPAZTXzgF84AzKdOAcgCsIZlh6BuAKC7wA9IsHBAB2AZ3h8AgmhBwAvClkAeKVCjIAnsqTAAJgC44EqMBEBzCgD4rACgDaAXQCU3foPjZREuHxLQ8BRx8Qn98W1cuIA

**Comment by dai-shi** [comment]
Thanks for reporting. I wasn't aware such a difference.

Hmm, not exporting `Store` type is intentional, but I see you have the issue with `--declaration`. Are you developing a library?

I'm not sure if we can go back to v2.8.x type if we do #2687.

Meanwhile, does this workaround work for you?
https://tsplay.dev/w2eA8W

**Reply by gitsunmin** [reply]
Thank you for your answer.

It's not even a library development, but the option 'declaration' was turned on in the project. I changed it to `false` and solved it. 😅

Thank you very much. 🙏

**Reply by dai-shi** [reply]
Good to hear.

> the option 'declaration' was turned on in the project.

That's interesting.
